test/test_python_dispatch.py

```diff
@@ -706,6 +706,27 @@ def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
         x.neg()
         self.assertEqual(called, [torch.ops.aten.neg.default])
 
+    def test_set_data(self):
+        called = 0
+
+        class SubTensor(torch.Tensor):
+            __torch_function__ = torch._C._disabled_torch_function_impl
+
+            @classmethod
+            def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
+                nonlocal called
+                called += 1
+                return super().__torch_dispatch__(func, types, args, kwargs)
+
+        x = SubTensor(torch.empty(2))
+        x.data
+        self.assertEqual(called, 1)
+        x.data = torch.empty(2)
```

Comment:
I would expect set_ to have the same behavior here (and also be fixed by this PR) could you add a test for that?

Reply:
I think set_ already worked, because it gets handled by torch dispatch already